Client/mods/deathmatch/logic/luadefs/CLuaEffectDefs.cpp

```diff
@@ -639,3 +642,12 @@ int CLuaEffectDefs::SetEffectDensity(lua_State* luaVM)
     lua_pushboolean(luaVM, false);
     return 1;
 }
+
+bool CLuaEffectDefs::FxCreateParticle(std::string strParticle, CVector vecPosition, CVector vecDirection, float fR, float fG, float fB, float fA, std::optional<bool> bRandomizeColors, std::optional<int> iCount, std::optional<float> fBrightness, std::optional<float> fSize, std::optional<bool> bRandomizeSizes, std::optional<float> fLife)
+{
+    eFxParticleSystems eParticleSystem;
+    if (!StringToEnum(strParticle, eParticleSystem))
```

Comment:
Did you try to substitute eFxParticleSystems directly into the function's arguments?